### ISSUE TITLE
Image no longer working after the last update

### ISSUE THREAD

**Issue by chitezh** [jshimko/meteor-launchpad#11]
--unsafe-perm: unknown option.
Usage: meteor build <output path> [--debug] [--directory] [--server-only]
          [--mobile-settings settings.json] [--server http://example.com:3000]

Package this project up for deployment. The command outputs a directory with
builds for all platforms in this project.

If you have added mobile platforms to your project with the
'meteor add-platform' command, then the output directory will contain
subdirectories named 'android' (with the APK bundle and Android project
source) and/or 'ios' (with the Xcode project source).

**Comment by jshimko** [comment]
What version of Meteor is your app using?

**Comment by chitezh** [comment]
METEOR@1.4.1.1

It was working some 40 minutes ago before your last push.

**Comment by jshimko** [comment]
This non-root restriction from Meteor 1.4.2 is killing me.

Give me a little bit.  I'm working on it...

**Comment by chitezh** [comment]
Alright. Thanks, waiting.

**Comment by jshimko** [comment]
Ok, pull and give it another try.

**Comment by chitezh** [comment]
Thanks, let me try

**Comment by chitezh** [comment]
Getting this error

```
Now you need to do one of the following:

  (1) Add "\/home/node/.meteor" to your path, or
  (2) Run this command as root:
        cp "/home/node/.meteor/packages/meteor-tool/1.4.1_1/mt-os.linux.x86_64/scripts/admin/launch-meteor" /usr/bin/meteor

Then to get started, take a look at 'meteor --help' or see the docs at
docs.meteor.com.
cp: cannot stat '/home/node/.meteor/packages/meteor-tool/1.4.1.1/mt-os.linux.x86_64/scripts/admin/launch-meteor': No such file or directory
```

**Comment by keyscores** [comment]
+1 seeing same error

**Comment by jshimko** [comment]
This should now be fixed with the latest build.  I tested with multiple versions of Meteor (including 1.4.1.1) and all is well.

Let me know if it does the trick for you.

**Comment by chitezh** [comment]
I am getting this error.

chown: invalid user: 'node' when I run the application after building

**Comment by jshimko** [comment]
Strange.  I'm _not_ getting that error in several different apps.  Do you definitely have the latest build?

``` sh
docker pull jshimko/meteor-launchpad:latest
```

**Comment by 0o-de-lally** [comment]
The build was passing for me yesterday. I mistakenly thought I saw the same
error.
Off topic: can I get your input here:
https://forums.meteor.com/t/docker-team-up-to-create-a-reference-docker-image-for-meteor/30829

Thanks

**Comment by jshimko** [comment]
I've reverted to building everything as root and then changing the permissions of the built app directory after everything is done.  It was proving to be too complicated to do all of the user switching in the build scripts.  In the end, the same thing has been accomplished... app files are owned by a non-root user and the Node process is started by that same user.

So give this another try and let me know if it clears everything up.  I've tested in several apps and haven't had any more issues.

(don't forget to `docker pull` the latest first)

**Comment by tnargwoxow** [comment]
I'm building straight through docker and also get the --unsafe-perm error.
I'm on Meteor 1.3.4.4 - Nice work though thanks a lot.

**Comment by jshimko** [comment]
Yeah, Meteor has been killing me with that for the last week.  Version 1.4.2 insists on that and earlier versions fails with it.  Pretty frustrating to support multiple Meteor versions at this point.  Pretty sure they're supposed to be pushing a fix for that soon though (so it warns instead of totally failing the build).  Until then, I think I need to leave that arg there so I can support the current release of Meteor.  I'll post here when things change though.

**Comment by jshimko** [comment]
FYI...

The open issue for this...
https://github.com/meteor/meteor/issues/7959

The 1.4.2.1 PR that should hopefully fix it...
https://github.com/meteor/meteor/pull/7975

The commit that will give us an environment variable to use so we don't break earlier Meteor versions with `--unsafe-perm`
https://github.com/meteor/meteor/pull/7975/commits/e4acc36f63f98243237b5e6b3e46b083822b95fd

**Comment by tnargwoxow** [comment]
Thanks for the info. How annoying!

**Comment by jykae** [comment]
@jshimko first of all, great image! Could you provide version tags for each release for docker image in Docker Hub. It would help to keep control if some update breaks things.

**Comment by xwilders** [comment]
I was able to deploy properly with Meteor 1.4.2 on Monday without hitting the root error (which felt great...).
However today I hit the root error again:
"You have run Meteor as root. Your permissions in your app directory will be incorrect..."

I thought this might be due to your recent changes, so I rebuilt an image based on the Monday version (which I had forked just in case), but the root error resurfaced.

Any ideas?

**Comment by jshimko** [comment]
@xwilders That's a warning, not an error.  The build should still finish and the script will then change the permissions of the app directory to a non-root user.  Trying to run `meteor build` as non-root was turning out to be a nightmare (depending on what npm packages you might have), so I reverted that part and instead just fix the permissions after the build.

Meteor is completely removed from the container after the build, so warnings about running Meteor as non-root later are irrelevant.  The app is run directly by Node using a non-root user.

**Comment by jshimko** [comment]
@jykae yes.  See https://github.com/jshimko/meteor-launchpad/issues/17

**Comment by jshimko** [comment]
I'm pretty sure the problems associated with this issue have been fixed for now, so I'm going to close this issue.  Feel free to open another more specific issue if any of you continue to run into problems.

Thanks!

**Comment by jykae** [comment]
@jshimko After meteor update from 1.4.1.1 to 1.4.2 success, earlier versions don't have option --unsafe-perm

**Comment by jshimko** [comment]
@jykae yes, that is unfortunately what we have to deal with until Meteor 1.4.2.1 gets released.  The problem has already been fixed in Meteor, but they're working on a few other fixes for 1.4.2 before pushing out the 1.4.2.1 release.  Until then, this image only officially supports 1.4.2.  Sorry for any inconvenience.  Once 1.4.2.1 hits, any app running 1.3 or higher should be supported.

See https://github.com/meteor/meteor/pull/7975

There's a release candidate now, so I'm assuming it should be releasing soon...
https://github.com/meteor/meteor/pull/7975#issuecomment-258215566
